packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -62,40 +59,14 @@ export interface GridRowProps extends React.HTMLAttributes<HTMLDivElement> {
   isFirstVisible: boolean;
   isLastVisible: boolean;
   isNotVisible: boolean;
+  showBottomBorder: boolean;
   onClick?: React.MouseEventHandler<HTMLDivElement>;
   onDoubleClick?: React.MouseEventHandler<HTMLDivElement>;
   onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
   onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
   [x: string]: any; // Allow custom attributes like data-* and aria-*
 }
 
-type OwnerState = Pick<GridRowProps, 'selected'> & {
-  editable: boolean;
-  editing: boolean;
-  isFirstVisible: boolean;
-  isLastVisible: boolean;
-  classes?: DataGridProcessedProps['classes'];
-  rowHeight: GridRowProps['rowHeight'];
-};
-
-const useUtilityClasses = (ownerState: OwnerState) => {
```

Comment:
I did a drive-by refactor here, not sure if it will be controversial. I was annoyed that I needed to add more stuff to the `type OwnerState` if I wanted to use variables internal to `GridRow`, so I removed this whole boilerplate section and replaced it with the `composeGridClasses` thing below that allows specifying utility classes inline. I could revert it if it's too controversial.

Reply:
Why do you think it's controversial? I like it, can't see any downsides here 👍🏻

Reply:
There is a very consistent convention throughout the company's codebase to use a split `useUtilityClasses` to define classes. Happy to see it's accepted.